### ISSUE TITLE
Multi-threading freezing main thread

### ISSUE THREAD

**Issue by DeltaNeverUsed** [NotNite/GDWeave#21]
if you try to start a new thread or use await anywhere, it'll freeze the main thread.

Minimal code to replicate:
```csharp
new Thread(() => { }).Start();
```
or in an async function
```csharp
await Task.Delay(1000);
```

**Comment by NotNite** [comment]
This is due to everyone's least favorite Win32 loader lock. I've had a plan to defeat it but it's extremely annoying so it might take me a bit.

**Comment by DeltaNeverUsed** [comment]
I see.
What was your plan to defeat it?
I wasn't aware of windows being like this, I ended up reading ["Perfect DLL Hijacking"](https://elliotonsecurity.com/perfect-dll-hijacking/) although a lot of it went over my head

**Comment by NotNite** [comment]
It was indeed that Perfect DLL Hijacking post, with the https://github.com/ElliotKillick/LdrLockLiberator repository, but it would require a lot of porting that C code. I might rewrite the loader in C# with NativeAOT to make it easier to port.

**Comment by DeltaNeverUsed** [comment]
Heya, just wanted to say that I think I have a working workaround, I need to test that it doesn't break anything else, but other than that it's looking pretty good.

**Comment by DeltaNeverUsed** [comment]
Okay, yeah, this still works!
I do also need to test under windows as well..

**Comment by DeltaNeverUsed** [comment]
Didn't mean to close the issue... that was a misclick- oops

**Comment by NotNite** [comment]
> Heya, just wanted to say that I think I have a working workaround, I need to test that it doesn't break anything else, but other than that it's looking pretty good.

I'll be happy to review a PR! I don't often mess with multithreading inside of a proxy DLL, so any help is appreciated. :)

**Comment by DeltaNeverUsed** [comment]
I'm not very experienced with rust, so things might be a little rough, but I'll make a PR in the next few hours.
I couldn't figure out how to intercept a call with that [proxy-dll](https://crates.io/crates/proxy-dll) crate you're using, so I ended up using [this crate](https://github.com/WarrenHood/proxygen).
But that crate wasn't up-to-date, so I had to modify it to work properly. I did that using the [patch.crates-io] thingy, but I'm not sure if that's the correct way to do it.
